tools/python/boutiques/localExec.py

```diff
@@ -132,8 +132,21 @@ def __init__(self, desc, invocation, options={}):
         self.desc_path = desc    # Save descriptor path
         self.errs = []        # Empty errors holder
         self.invocation = invocation
-        # Parse JSON descriptor
-        self.desc_dict = loadJson(desc)
+
+        # Extra Options
+        # Include: forcePathType and debug
+        self.debug = False
+        self.zenodo = False
+        for option in list(options.keys()):
+            setattr(self, option, options.get(option))
+
+        if self.zenodo:
+            from boutiques.puller import Puller
+            puller = Puller(desc, self.debug, False)
+            self.desc_dict = json.loads(puller.pull().read())
```

Comment:
We should be careful with reproducibility here. We need to make sure that users won't be running a different analysis if a new descriptor is pushed between two runs. My feeling is that Zenodo ids are immutable, i.e., publishing a new version of the descriptor would result in a new id. Is it indeed the case? 

Note: Docker container images are not immutable, one might still mess up reproducibility that way, but we shouldn't add to this issue here.

Reply:
Yes, a new Zenodo ID is generated every time you publish a new version.